### PR TITLE
m3u8 reorder task

### PR DESCRIPTION
**Pull request by abeglova**
#### What are the relevant tickets?
https://github.com/mitodl/odl-video-service/issues/810
#### What's this PR do?
This pr reorders m3u8 playlists (video__index.m3u8 on S3) from highest bandwidth to lowest bandwidth so that the highest bandwidth video is selected first.
#### How should this be manually tested?
Set 
ET_PRESET_IDS=1351620000001-200050,1351620000001-200040,1504127981769-6cnqhq,1504127981819-v44xlx,1504127981867-06dkm6,1504127981921-c2jlwt

in your .env file.  Upload a video. Check that the newly created video__index.m3u8 file in {VIDEO_S3_TRANSCODE_BUCKET}/transcoded/{video_key} s3 folder has the smallest bandwidth first.

From the python shell, run

```
from cloudsync.tasks import *
sort_transcoded_m3u8_files()
```

The file on s3 should be updated and should now be sorted to have the highest  bandwidth first.

#### Where should the reviewer start?
(Optional)

#### Any background context you want to provide?
(Optional)

#### Screenshots (if appropriate)
(Optional)

#### What GIF best describes this PR or how it makes you feel?
(Optional)

### CODE REVIEW

File: cloudsync/tasks.py [hunk 1]
```diff
@@ -352,3 +352,37 @@ def parse_content_metadata(response):
...
+            lines = file_content.split(delimiter)
+            header = lines.pop(0)
+
+            if str.strip(header) != "#EXTM3U":
+                continue
+
+            try:
+                lines.sort(key=lambda line: -int(re.search(r'RESOLUTION=(\d+)', line).group(1)))
+            except AttributeError:
+                continue
```
Comment (mbertrand) [inline cloudsync/tasks.py]: Maybe add a log statement here to give an indication of what happened, or will this be a common noisy error?

File: cloudsync/tasks.py [hunk 2]
```diff
@@ -352,3 +352,37 @@ def parse_content_metadata(response):
...
+
+            delimiter = '#EXT-X-STREAM-INF:'
+            lines = file_content.split(delimiter)
+            header = lines.pop(0)
+
+            if str.strip(header) != "#EXTM3U":
+                continue
+
+            try:
+                lines.sort(key=lambda line: -int(re.search(r'RESOLUTION=(\d+)', line).group(1)))
```
Comment (mbertrand) [inline cloudsync/tasks.py]: Might also need to sort by bandwidth, there are two codecs with the same resolution but one has a lower bandwidth than the other:
```
#EXT-X-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=471000,RESOLUTION=320x176,CODECS="avc1.42001e,mp4a.40.2"
video_1351620000001-200050.m3u8
#EXT-X-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=682000,RESOLUTION=320x176,CODECS="avc1.42001e,mp4a.40.2"
video_1351620000001-200040.m3u8
```

After reordering, the lower bandwidth codec precedes the higher bandwidth code in some videos but not others

File: cloudsync/tasks.py (hunk 1 above)
Comment (abeglova) [reply cloudsync/tasks.py]: I don't think this will be a common error, I adding a log statement

**Comment by mbertrand** [comment]
PS A benefit of chunking out the videos would be we could process a subset of videos too, instead of all of them, for testing purposes.

Something like:
```python
@app.task(bind=True)
def sort_video_indexes(self):
    index_tasks = celery.group(
            [
                sort_transcoded_m3u8_files.si(video_ids)
                for video_ids in chunks(
                    Video.objects.order_by("id").values_list("id", flat=True),
                    chunk_size=settings.ELASTICSEARCH_INDEXING_CHUNK_SIZE,
                )
            ]
    .....


@app.task(bind=True)
def sort_transcoded_m3u8_files(video_ids):
    for video in Video.objects.filter(id__in=video_ids).iterator():
        ....
```

**Comment by mbertrand** [comment]
One way to reduce the number of videos processed on production is to filter by those that have an HLS encoding:

`for video in Video.objects.filter(videofile__encoding=EncodingNames.HLS).iterator():`

**Pull request merged**